app/main.py

```diff
@@ -1,3 +1,68 @@
+from __future__ import annotations
+from typing import Union
+
+
 class Distance:
-    # Write your code here
-    pass
+    Variables = Union["Distance", int, float]
+
+    def __init__(self, km: float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Variables) -> Distance:
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        if isinstance(other, (float, int)):
+            return Distance(self.km + other)
+
+    def __iadd__(self, other: Variables) -> Distance:
+        if isinstance(other, Distance):
+            self.km += other.km
+        elif isinstance(other, (float, int)):
+            self.km += other
+        return self
+
+    def __mul__(self, other: Variables) -> Distance:
+        if isinstance(other, float | int):
```

Comment:
The use of the `|` operator for type checking is only available in Python 3.10 and later. Since the file imports `annotations` from `__future__`, it suggests compatibility with earlier versions. Consider using `isinstance(other, (float, int))` instead.